build/build/src/engine.rs

```diff
@@ -206,13 +213,36 @@ impl BuildConfigurationFlags {
     pub fn build_launcher_package(&self) -> bool {
         self.build_launcher_package || self.build_launcher_bundle
     }
+
+    pub fn add_standard_library_test_selection(&mut self, selection: StandardLibraryTestsSelection) {
+        match selection {
+            StandardLibraryTestsSelection::All => {
+                self.test_standard_library = Some(StandardLibraryTestsSelection::All);
+            }
+            StandardLibraryTestsSelection::Selected(new_selection) => {
+                match &mut self.test_standard_library {
+                    Some(StandardLibraryTestsSelection::All) => {
```

Comment:
This could be simplified to one-level match on a tuple.
```rs
	use StandardLibraryTestsSelection::*;
        let combined_selection = match (self.test_standard_library.take(), selection) {
	    (None, selection) => selection,
            (Some(All), _) | (_, All) => All,
	    (Some(Selected(selection), Selected(new_selection)) => {
		selection.extend(new_selection);
		Selected(selection)
	    }
	}
	self.test_standard_library = Some(combined_selection);
```

Reply:
Thanks for the suggestion, I will add it.